linkerd/app/inbound/src/direct.rs

```diff
@@ -70,6 +70,8 @@ where
     let http = svc::stack(http_loopback)
         .push_on_response(
             svc::layers()
+                .push(svc::layer::mk(svc::SpawnReady::new))
+                .push(metrics.stack.layer(crate::stack_labels("http", "gateway")))
```

Comment:
it occurs to me that since the direct stack uses the same TCP forward stack as the normal inbound router, the TCP stack metrics won't have any special labels indicating that they're the TCP gateway rather than in-cluster TCP inbound. It might be a significantly bigger change to allow passing in the tcp forward stack but modifying its labels, though, so maybe that's best saved for a follow-up...

Reply:
There is no TCP gateway yet. It's all TCP inbound. Will change soon.